compiler-core/src/type_/environment.rs

```diff
@@ -503,6 +505,17 @@ impl<'a> Environment<'a> {
             self.warnings
                 .emit(Warning::UnusedImportedModule { name, location });
         }
+
+        for (alias, location) in self.unused_module_aliases.clone().into_iter() {
+            if let Some((_, module)) = self.imported_modules.get(&alias) {
+                if let Some(name) = module.name.split('/').last() {
+                    if self.imported_modules.get(name).is_none() {
```

Comment:
This is going to not raise an unused import warning if there's another module with a similar name. Another import doesn't stop the other one from being unused, so we want to emit the warning here.